Code/Tools/AssetProcessor/native/utilities/BuilderList.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/std/string/string.h>
+#include <utilities/Builder.h>
+
+namespace AssetProcessor
+{
+    //! Helper class that keeps track of the Builders and manages reserving a builder specifically for CreateJobs
+    //! This class is not inherently thread-safe and must be locked before any access
+    class BuilderList
+    {
+    public:
+        BuilderList() = default;
+
+        void AddBuilder(AZStd::shared_ptr<Builder> builder, BuilderPurpose purpose);
```

Comment:
Consider adding doc strings here for public APIs

Reply:
This is pretty much intended to be a private class.  Its just split out into a different file because BuilderManager is getting way too big.  What's the standard for documenting in this case?